website/docs/d/connect_instance_storage_config.html.markdown

```diff
@@ -0,0 +1,81 @@
+---
+subcategory: "Connect"
+layout: "aws"
+page_title: "AWS: aws_connect_instance_storage_config"
+description: |-
+  Provides details about a specific Amazon Connect Instance Storage Config.
+---
+
+# Data Source: aws_connect_instance_storage_config
+
+Provides details about a specific Amazon Connect Instance Storage Config.
+
+## Example Usage
+
+```hcl
+data "aws_connect_instance_storage_config" "example" {
+  association_id = "1234567890123456789012345678901234567890123456789012345678901234"
+  instance_id    = "aaaaaaaa-bbbb-cccc-dddd-111111111111"
+  resource_type  = "CONTACT_TRACE_RECORDS"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `association_id` - (Required) The existing association identifier that uniquely identifies the resource type and storage config for the given instance ID.
+* `instance_id` - (Required) Reference to the hosting Amazon Connect Instance
+* `resource_type` - (Required) A valid resource type. Valid Values: `CHAT_TRANSCRIPTS` | `CALL_RECORDINGS` | `SCHEDULED_REPORTS` | `MEDIA_STREAMS` | `CONTACT_TRACE_RECORDS` | `AGENT_EVENTS` | `REAL_TIME_CONTACT_ANALYSIS_SEGMENTS`.
+
+## Attributes Reference
+
+In addition to all of the arguments above, the following attributes are exported:
+
+* `id` - The identifier of the hosting Amazon Connect Instance, `association_id`, and `resource_type` separated by a colon (`:`).
```

Comment:
Are `id` and `instance_id` the same value? Or is `id` a unique value? It's not clear to me from this description.

Reply:
The `id` is formed by `<instance_id >:<association_id>:<resource_type>` where `instance_id` is part of the formed `id`. `instance_id` is described in full as `The identifier of the hosting Amazon Connect Instance`. Would it be clearer for practitioners if it's changed to 

```suggestion
* `id` - Formed from the `instance_id`, `association_id`, and `resource_type` separated by a colon (`:`).

Reply:
`id` is technically a leaked implementation detail from `terrraform-plugin-sdk`, so documenting it is not required.